frontends/web/src/routes/settings/electrum-add-server.tsx

```diff
@@ -0,0 +1,145 @@
+import style from './electrum.module.css';
+
+type Props = {
+  onAdd: (server: TElectrumServer) => void;
+};
+
+export const ElectrumAddServer = ({
+  onAdd,
+}: Props) => {
+  const { t } = useTranslation();
+  const [valid, setValid] = useState<boolean>();
+  const [electrumServer, setElectrumServer] = useState<string>('');
+  const [electrumCert, setElectrumCert] = useState<string>('');
+  const [loadingCheck, setLoadingCheck] = useState<boolean>();
+  const [loadingCert, setLoadingCert] = useState<boolean>();
+
+  const getServer = (): TElectrumServer => {
+    return {
+      server: electrumServer.trim(),
+      pemCert: electrumCert,
+      tls: electrumCert !== '',
+    };
+  };
+
+  const add = () => {
+    onAdd(getServer());
+    setElectrumServer('');
+    setElectrumCert('');
+  };
+
+  const downloadCert = async () => {
+    setLoadingCert(true);
+    const data = await apiPost('certs/download', electrumServer.trim());
+    if (data.success) {
+      setElectrumCert(data.pemCert);
+    } else {
+      alertUser(data.errorMessage);
+    }
+    setLoadingCert(false);
+  };
+
+  const check = async () => {
+    setLoadingCheck(true);
+    const { success, errorMessage } = await apiPost('electrum/check', getServer());
+    if (success) {
+      alertUser(t('settings.electrum.checkSuccess', { host: electrumServer }));
+    } else {
+      alertUser(t('settings.electrum.checkFailed') + ':\n' + errorMessage);
+    }
+    setValid(success);
+    setLoadingCheck(false);
+  };
+
+  return (
+    <div className={style.addServer}>
+      <div className="flex flex-row flex-start flex-wrap">
+        <p className={style.badge}>{t('settings.electrum.step1')}</p>
+        <div className="flex-1">
+          <p>{t('settings.electrum.step1-text')}</p>
+        </div>
+      </div>
+      <Input
+        name="electrumServer"
+        onInput={event => setElectrumServer(event.target.value)}
+        value={electrumServer}
+        placeholder="host:port"
+      />
+      <div className="flex flex-row flex-start flex-wrap">
+        <p className={style.badge}>{t('settings.electrum.step2')}</p>
+        <div className="flex-1">
+          <p>{t('settings.electrum.step2-text')}</p>
+          <p>{t('settings.electrum.step2-text-tcp')}</p>
+        </div>
+      </div>
+      <textarea
+        className={style.textarea}
+        rows={10}
+        cols={80}
+        name="electrumCert"
+        onInput={event => setElectrumCert((event.target as HTMLTextAreaElement).value)}
+        value={electrumCert}
+        placeholder={'-----BEGIN CERTIFICATE-----\n...\n-----END CERTIFICATE-----'}
+      />
+      <div className={[style.block, 'flex flex-row flex-end'].join(' ')}>
+        <Button primary disabled={loadingCert || electrumCert !== ''} onClick={downloadCert}>
+          {
+            loadingCert && (
+              <div className={style.miniSpinnerContainer}>
+                <div className={style.miniSpinner}></div>
+              </div>
```

Comment:
As this exact same "loading" component is being used in multiple places (twice in this file, and once in `electrum-server`), it makes sense to extract it into a different file / component.

Reply:
Good suggestion - will maybe try that in another PR

Reply:
Hmm it doesn't work that easily, the buttons have different styles.